Feat(deployment) cloud-unit config for running BHIMA on a linux server

jeremielodi opened the pull request:
This PR make  the depoyment of BHIMA easy thanks to the cloud-unit configuration file that we can just copy and paste while creating a server on line.
We tested this script on DigitalOcean and it works flawlessly.

closes #3620

jeremielodi committed:
Feat(deployment) cloud-config for running BHIMA on a lunix server

closes #3620

mbayopanda commented on docs/en/install/index.md:
```diff
@@ -0,0 +1,108 @@
+</li>
+</ul>
+
+# Launch BHIMA to the broser
```
Launch BHIMA to the browser

jniles commented on docs/cloud-init.yml:
```diff
@@ -0,0 +1,129 @@
+packages:
+  - curl
+  - nginx
+  - git
```
Looks like we don't use git anymore - see if we can remove this and it still pass successfully.

jniles commented on docs/cloud-init.yml:
```diff
@@ -0,0 +1,129 @@
+  - build-essential
+  - redis-server
+  - redis-tools
+  - screen
```
I added screen during my testing, but it really isn't necessary.  Let's take it out.

jniles commented on docs/cloud-init.yml:
```diff
@@ -0,0 +1,129 @@
+      FLUSH PRIVILEGES;
+  - path: /tmp/bhima-latest-release.sh
+    content: |
+      wget -O bhima-latest $(curl -s https://api.github.com/repos/IMA-WorldHealth/bhima/releases/latest | jq -r '.zipball_url');
```
Beautiful :+1:

jniles commented on docs/en/install/index.md:
```diff
@@ -0,0 +1,108 @@
+</p>
+
+### Tip
+If you have not already created a DigitalOcean account, use our referral link to do so: https://m.do.co/c/39937689124c.
```
Technically, this isn't our referral code link.  We can either put our own (generated by DO) or note that this is the Open Data Kit's referral link.  Can you do that?

jniles commented on docs/fr/install/index.md:
````diff
@@ -0,0 +1,108 @@
+</li>
+</ul>
+
+3. Une fois connecté, exécutez la commande ``` sudo certbot run --nginx --non-interactive --agree-tos -m YOUR_EMAIL --redirect -d YOUR_DOMAIN.```
````
Note that this will only work if they change the [`server_name`](https://github.com/IMA-WorldHealth/bhima/pull/3987/files#diff-aad866a9311fd8ed6d4c5e99baf24a1eR56) variable in the nginx configuration from `demo.bhi.ma` to whatever their domain name is.  We should probably write that somewhere.

jniles reviewed (commented):
Just a few comments here.

mbayopanda commented on docs/cloud-init.yml:
```diff
@@ -0,0 +1,129 @@
+  - yarn --ignore-engines
+  - NODE_ENV=production yarn build
+  - cd bin
+  - NODE_ENV=production pm2 start server/app.js -i 1 --name "bhima"
```
Could you add `pm2 startup` and `pm2 save` commands, to startup the application automatically when the computer reboot

mbayopanda reviewed (commented):
We need to handle startup on reboot with `pm2 startup` and `pm2 save`

jniles committed:
chore: add gzip and use systemd to monitor deploy

Ensures that gzip is applied to all css/js assets and uses systemd to
control the deployment of the application under the user bhima.

jniles committed:
Merge remote-tracking branch 'upstream/master' into cloud_init
